daemons/mrpd/msrp.c

```diff
@@ -628,6 +628,9 @@ int msrp_event(int event, struct msrp_attribute *rattrib)
 #if LOG_MSRP
 			msrp_print_debug_info(event, attrib);
 #endif
+		} else {
+			/* free this attrib if we are not interested */
+			free(rattrib);
```

Comment:
This is my first look at the mrpd code so this might be a common pattern and handled properly with an overloaded free() or something I am missing. The allocation pointed to by rattrib is freed but the pointer is never nulled out. Have concerns if it could be used later. However, I do see that in this case rattrib is passed into this function as a simple pointer so nulling it won't do any good outside of this function scope. The other concern is that on a quick search for this function msrp_event() that the caller of this function can also free this same allocation that was freed inside msrp_event() (at least in the one call case that I looked at). Concern would be a double free and potentially freeing some new allocation that might have occurred since the first free() or other general memory corruption.